File: Lecture 6/decimalToBinary.cpp
#include <iostream>
using namespace std;
int decimalToBinary(int decNum){
    int ans = 0, pow = 1;
    while(decNum>0){
        int rem = decNum % 2;
        decNum /= 2;                //Update
        ans += (rem * pow);      //Update
        pow *= 10;                  

    }
    return ans;                     //binary form
}

//binary to decimal
int binaryToDecimal(int binNum){
    int ans = 0, pow = 1;
    while(binNum > 0){
        int rem = binNum % 10;
        ans += (rem * pow);

        binNum /= 10;
        pow *= 2;
    }
    return ans;
}
int main(){
    int decNum, binNum;
    cout << "Enter a decimal number: ";
    cin >> decNum;
    cout << "Enter a binary number: ";
    cin >> binNum;
    cout << "Binary number of " << decNum << " is: " << decimalToBinary(decNum) << endl;
    cout << "Decimal number of " << binNum << " is: " << binaryToDecimal(binNum) << endl;

    return 0;
}

File: Lecture 5 Python/decimalToBinary.py
# Decimal To Binary 
# DecNum = int(input("Enter a Decimal Number: "))
# pow = 1
# ans = 0
# while DecNum > 0:
#     remainder = DecNum % 2
#     DecNum = DecNum // 2
#     ans += (remainder * pow)
#     pow *= 10
# print(f"Binary Number of {DecNum} is: ", ans)


# Binary To Decimal
# binNum = int(input("Enter a Binary Number: "))
# pow = 1
# ans = 0
# while(binNum > 0):
#     remainder = binNum % 10
#     ans += remainder * pow
#     binNum //= 10
#     pow *= 2
# print(f"Binary of {binNum} is: ", ans)

# # ✅ Decimal to Binary Conversion
# DecNum = int(input("Enter a Decimal Number: "))
# power = 1
# binary = 0
# original = DecNum  # Save for output

# while DecNum > 0:
#     remainder = DecNum % 2
#     DecNum //= 2
#     binary += remainder * power
#     power *= 10

# print(f"Binary of {original} is: {binary}")

# ✅ Binary to Decimal Conversion
binNum = int(input("Enter a Binary Number: "))
power = 1
decimal = 0
original = binNum  # Save for output

while binNum > 0:
    remainder = binNum % 10
    decimal += remainder * power
    binNum //= 10
    power *= 2

print(f"Decimal of {original} is: {decimal}")
